sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -80,6 +115,8 @@ internal class NextLinkOperationImplementation : IOperation
             OperationFinalStateVia finalStateVia,
             string? apiVersion)
         {
+            Argument.AssertNotNull(pipeline, nameof(pipeline));
```

Comment:
Would it make sense to validate other arguments for null?  Nullability annotations aren't 100% failsafe -- someone could still pass null for these with the null-forgiving operator.

Reply:
You are right, will add the rest.